<a href="https://colab.research.google.com/github/soadD/data_model/blob/main/Projeto_Modelo_de_An%C3%A1lise_de_Dados.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Análise de dados utilizando a API nba_api

Utilizarei este projeto para exercitar meus conhecimentos em Data Analysis e Data scraping e assim fazer um passo a passo com os códigos mais utilizados no processo de extração, análise e carregamento de dados.


## Extração dos Dados

In [1]:
!pip install nba_api

In [2]:
from nba_api.stats.static import players
import pandas as pd

### Verificando se a importação foi bem sucedida

In [3]:
jog_nba = players.get_players()
print('Conferindo o número total de jogadores:{}'.format(len(jog_nba)))

Conferindo o número total de jogadores:4900


### Procurando por um jogador específico na API - Donovan Mitchell

In [4]:
# O objeto players foi importado pela API da nba e também o método get_players, que traz um dicionário com as informações básicas dos jogadores
jog_nba = players.get_players()
'''def get_players():
    players_list = []
    for player in players:
        players_list.append(_get_player_dict(player))
    return players_list
    '''
#Trouxe todos os meus jogadores da lista players para a minha lista jog_nba

spida = []
#Rodando um for procurando pelo jogador específico
for player in jog_nba:
    if player["full_name"] == 'Donovan Mitchell':
        spida = player
        break

In [5]:
print(spida)
spida_id = spida['id']


{'id': 1628378, 'full_name': 'Donovan Mitchell', 'first_name': 'Donovan', 'last_name': 'Mitchell', 'is_active': True}


Jogador encontrado agora temos seu ID = 1628378 e assim acessar alguns endpoints da api e avaliar seus dados

In [6]:
from nba_api.stats.endpoints import playergamelog

gamelog_spida = playergamelog.PlayerGameLog(spida_id,)
type(gamelog_spida)

nba_api.stats.endpoints.playergamelog.PlayerGameLog

Conseguimos importar as informações do jogador porém seu tipo ainda está como um objeto de API, hora de fazermos algumas transformações.

## Transformação

Para isso utilizamos o método get_data_frames para converter o objeto em df e adicionamos o índice 0 pois quando foi feita a extração pela primeira vez,
o objeto saia como uma lista, lendo a documentação parece que alguns endpoints são listas com varios dataframes, voce tem que selecionar o indice que irá
utilizar para extrair

In [7]:
import pandas as pd

In [8]:
df_spida = gamelog_spida.get_data_frames()[0]

In [9]:
df_spida

SEASON_ID  Player_ID     Game_ID     GAME_DATE      MATCHUP WL  MIN  FGM  \
0      22023    1628378  0022301174  APR 12, 2024  CLE vs. IND  W   38   12   
1      22023    1628378  0022301158  APR 10, 2024  CLE vs. MEM  W   34    9   
2      22023    1628378  0022301127  APR 06, 2024    CLE @ LAL  L   38    4   
3      22023    1628378  0022301110  APR 03, 2024    CLE @ PHX  L   38    8   
4      22023    1628378  0022301077  MAR 31, 2024    CLE @ DEN  L   31    3   
5      22023    1628378  0022301067  MAR 29, 2024  CLE vs. PHI  W   32    4   
6      22023    1628378  0022300967  MAR 16, 2024    CLE @ HOU  L   26    4   
7      22023    1628378  0022300951  MAR 13, 2024    CLE @ NOP  W   28    5   
8      22023    1628378  0022300846  FEB 28, 2024    CLE @ CHI  L   44    5   
9      22023    1628378  0022300832  FEB 27, 2024  CLE vs. DAL  W   38   11   
10     22023    1628378  0022300820  FEB 25, 2024    CLE @ WAS  W   36    4   
11     22023    1628378  0022300781  FEB 14, 2024  CLE vs. CHI  W   39    8   
12     22023    1628378  0022300762  FEB 12, 2024  CLE vs. PHI  L   37    9   
13     22023    1628378  0022300756  FEB 10, 2024    CLE @ TOR  W   31    5   
14     22023    1628378  0022300735  FEB 08, 2024    CLE @ BKN  W   30   10   
15     22023    1628378  0022300728  FEB 07, 2024    CLE @ WAS  W   38   14   
16     22023    1628378  0022300714  FEB 05, 2024  CLE vs. SAC  W   33   11   
17     22023    1628378  0022300703  FEB 03, 2024    CLE @ SAS  W   35   12   
18     22023    1628378  0022300687  FEB 01, 2024    CLE @ MEM  W   33   10   
19     22023    1628378  0022300675  JAN 31, 2024  CLE vs. DET  W   35   14   
20     22023    1628378  0022300658  JAN 29, 2024  CLE vs. LAC  W   39   11   
21     22023    1628378  0022300639  JAN 26, 2024    CLE @ MIL  W   37   12   
22     22023    1628378  0022300624  JAN 24, 2024    CLE @ MIL  L   39   10   
23     22023    1628378  0022300607  JAN 22, 2024    CLE @ ORL  W   30    9   
24     22023    1628378  0022300595  JAN 20, 2024    CLE @ ATL  W   36    6   
25     22023    1628378  0022300572  JAN 17, 2024  CLE vs. MIL  W   26   12   
26     22023    1628378  0022300561  JAN 15, 2024  CLE vs. CHI  W   37   11   
27     22023    1628378  0022300527  JAN 11, 2024  CLE vs. BKN  W   41   15   
28     22023    1628378  0022300497  JAN 07, 2024  CLE vs. SAS  W   37    7   
29     22023    1628378  0022300482  JAN 05, 2024  CLE vs. WAS  W   26    8   
30     22023    1628378  0022300465  JAN 03, 2024  CLE vs. WAS  W   24    6   
31     22023    1628378  0022300452  JAN 01, 2024    CLE @ TOR  L   38    8   
32     22023    1628378  0022300433  DEC 29, 2023  CLE vs. MIL  L   36   14   
33     22023    1628378  0022300349  DEC 18, 2023  CLE vs. HOU  W   45   14   
34     22023    1628378  0022300336  DEC 16, 2023  CLE vs. ATL  W   41    7   
35     22023    1628378  0022300319  DEC 14, 2023    CLE @ BOS  L   40   12   
36     22023    1628378  0022300305  DEC 12, 2023    CLE @ BOS  L   35   11   
37     22023    1628378  0022300294  DEC 11, 2023    CLE @ ORL  L   37    6   
38     22023    1628378  0022301220  DEC 08, 2023    CLE @ MIA  W   36    9   
39     22023    1628378  0022301205  DEC 06, 2023  CLE vs. ORL  W   37   12   
40     22023    1628378  0022300283  DEC 02, 2023    CLE @ DET  W   33    7   
41     22023    1628378  0022300265  NOV 30, 2023  CLE vs. POR  L   34    7   
42     22023    1628378  0022300055  NOV 28, 2023  CLE vs. ATL  W   35   14   
43     22023    1628378  0022300251  NOV 26, 2023  CLE vs. TOR  W   34    4   
44     22023    1628378  0022300242  NOV 25, 2023  CLE vs. LAL  L   35    4   
45     22023    1628378  0022300199  NOV 15, 2023    CLE @ POR  W   36   13   
46     22023    1628378  0022300191  NOV 13, 2023    CLE @ SAC  L   38    7   
47     22023    1628378  0022300176  NOV 11, 2023    CLE @ GSW  W   36    7   
48     22023    1628378  0022300167  NOV 08, 2023    CLE @ OKC  L   37    8   
49     22023    1628378  0022300142  NOV 05, 2023  CL

In [10]:
df_spida.count()
#Só vemos 55 jogos, este df serve apenas para o ano de 2023. Como eu não coloquei nenhum argumento de temporada na linha 14, e é um parâmetro obrigatório a função puxou o ano de 2023 por default

SEASON_ID          55
Player_ID          55
Game_ID            55
GAME_DATE          55
MATCHUP            55
WL                 55
MIN                55
FGM                55
FGA                55
FG_PCT             55
FG3M               55
FG3A               55
FG3_PCT            55
FTM                55
FTA                55
FT_PCT             55
OREB               55
DREB               55
REB                55
AST                55
STL                55
BLK                55
TOV                55
PF                 55
PTS                55
PLUS_MINUS         55
VIDEO_AVAILABLE    55
dtype: int64

## Extraindo SeasonAll values

Para buscar as estatísticas históricas de um jogador, precisamos importar mais alguns dados da API, ativando a função SeasonAll

In [11]:
from nba_api.stats.library.parameters import SeasonAll
from nba_api.stats.static import players

In [12]:
#Extraindo os dados gerais de Donovan Mitchell e transformando em dataframe
spida_all = playergamelog.PlayerGameLog(spida_id,season=SeasonAll.all)
spida_all = spida_all.get_data_frames()[0]

spida_all

SEASON_ID  Player_ID     Game_ID     GAME_DATE      MATCHUP WL  MIN  FGM  \
0       22023    1628378  0022301174  APR 12, 2024  CLE vs. IND  W   38   12   
1       22023    1628378  0022301158  APR 10, 2024  CLE vs. MEM  W   34    9   
2       22023    1628378  0022301127  APR 06, 2024    CLE @ LAL  L   38    4   
3       22023    1628378  0022301110  APR 03, 2024    CLE @ PHX  L   38    8   
4       22023    1628378  0022301077  MAR 31, 2024    CLE @ DEN  L   31    3   
..        ...        ...         ...           ...          ... ..  ...  ...   
463     22017    1628378  0021700062  OCT 25, 2017    UTA @ PHX  L   13    1   
464     22017    1628378  0021700054  OCT 24, 2017    UTA @ LAC  L   30    7   
465     22017    1628378  0021700036  OCT 21, 2017  UTA vs. OKC  W   22    1   
466     22017    1628378  0021700023  OCT 20, 2017    UTA @ MIN  L   20    1   
467     22017    1628378  0021700010  OCT 18, 2017  UTA vs. DEN  W   26    3   

     FGA  FG_PCT  ...  DREB  REB  AST  STL  BLK  TOV  PF  PTS  PLUS_MINUS  \
0     27   0.444  ...     2    4    5    4    0    1   2   33          12   
1     17   0.529  ...     3    4    8    3    1    3   2   29           2   
2     13   0.308  ...     5    5    7    3    2    5   2   10          10   
3     17   0.471  ...     3    3    2    2    2    1   1   24          -8   
4     12   0.250  ...     0    0    6    1    0    2   2   13         -27   
..   ...     ...  ...   ...  ...  ...  ...  ...  ...  ..  ...         ...   
463    7   0.143  ...     1    1    1    1    1    3   3    2          -2   
464   20   0.350  ...     1    2    1    1    0    4   4   19          -5   
465    7   0.143  ...     2    2    6    1    0    0   4    2           3   
466    7   0.143  ...     0    0    2    0    2    1   2    2           3   
467   11   0.273  ...     1    1    4    0    1    3   1   10          22   

     VIDEO_AVAILABLE  
0                  1  
1                  1  
2                  1  
3                  1  
4                  1  
..               ...  
463                1  
464                1  
465                1  
466                1  
467                1  

[468 rows x 27 columns]

In [13]:
#Verificando Dataframe
type(spida_all)

pandas.core.frame.DataFrame

## Transformação

In [14]:
# Para melhor visualização dos dados, vamos permitir a visualização de todas as colunas
pd.set_option('display.max_columns', 100)

spida_all

SEASON_ID  Player_ID     Game_ID     GAME_DATE      MATCHUP WL  MIN  FGM  \
0       22023    1628378  0022301174  APR 12, 2024  CLE vs. IND  W   38   12   
1       22023    1628378  0022301158  APR 10, 2024  CLE vs. MEM  W   34    9   
2       22023    1628378  0022301127  APR 06, 2024    CLE @ LAL  L   38    4   
3       22023    1628378  0022301110  APR 03, 2024    CLE @ PHX  L   38    8   
4       22023    1628378  0022301077  MAR 31, 2024    CLE @ DEN  L   31    3   
..        ...        ...         ...           ...          ... ..  ...  ...   
463     22017    1628378  0021700062  OCT 25, 2017    UTA @ PHX  L   13    1   
464     22017    1628378  0021700054  OCT 24, 2017    UTA @ LAC  L   30    7   
465     22017    1628378  0021700036  OCT 21, 2017  UTA vs. OKC  W   22    1   
466     22017    1628378  0021700023  OCT 20, 2017    UTA @ MIN  L   20    1   
467     22017    1628378  0021700010  OCT 18, 2017  UTA vs. DEN  W   26    3   

     FGA  FG_PCT  FG3M  FG3A  FG3_PCT  FTM  FTA  FT_PCT  OREB  DREB  REB  AST  \
0     27   0.444     4    10    0.400    5    5   1.000     2     2    4    5   
1     17   0.529     5    10    0.500    6    7   0.857     1     3    4    8   
2     13   0.308     2     9    0.222    0    2   0.000     0     5    5    7   
3     17   0.471     2     7    0.286    6    6   1.000     0     3    3    2   
4     12   0.250     2     6    0.333    5    5   1.000     0     0    0    6   
..   ...     ...   ...   ...      ...  ...  ...     ...   ...   ...  ...  ...   
463    7   0.143     0     2    0.000    0    0   0.000     0     1    1    1   
464   20   0.350     3     9    0.333    2    2   1.000     1     1    2    1   
465    7   0.143     0     3    0.000    0    0   0.000     0     2    2    6   
466    7   0.143     0     2    0.000    0    1   0.000     0     0    0    2   
467   11   0.273     0     2    0.000    4    4   1.000     0     1    1    4   

     STL  BLK  TOV  PF  PTS  PLUS_MINUS  VIDEO_AVAILABLE  
0      4    0    1   2   33          12                1  
1      3    1    3   2   29           2                1  
2      3    2    5   2   10          10                1  
3      2    2    1   1   24          -8                1  
4      1    0    2   2   13         -27                1  
..   ...  ...  ...  ..  ...         ...              ...  
463    1    1    3   3    2          -2                1  
464    1    0    4   4   19          -5                1  
465    1    0    0   4    2           3                1  
466    0    2    1   2    2           3                1  
467    0    1    3   1   10          22                1  

[468 rows x 27 columns]

In [15]:
# Informações gerais do dataset
spida_all.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 468 entries, 0 to 467
Data columns (total 27 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   SEASON_ID        468 non-null    object 
 1   Player_ID        468 non-null    int64  
 2   Game_ID          468 non-null    object 
 3   GAME_DATE        468 non-null    object 
 4   MATCHUP          468 non-null    object 
 5   WL               468 non-null    object 
 6   MIN              468 non-null    int64  
 7   FGM              468 non-null    int64  
 8   FGA              468 non-null    int64  
 9   FG_PCT           468 non-null    float64
 10  FG3M             468 non-null    int64  
 11  FG3A             468 non-null    int64  
 12  FG3_PCT          468 non-null    float64
 13  FTM              468 non-null    int64  
 14  FTA              468 non-null    int64  
 15  FT_PCT           468 non-null    float64
 16  OREB             468 non-null    int64  
 17  DREB            

In [16]:
#O dataset não possui nenhum valor nulo e aparentemente todas as colunas estão com tipos adequados

### Eliminando colunas desnecessárias

In [17]:
# Verificação de valores únicos
print(len(spida_all.Game_ID.unique()))
print(len(spida_all.Player_ID.unique()))

468
1


In [18]:
#O Player ID é o mesmo por todo o dataset, diferentemente do Game ID que é único para cada partida que este jogador participou. Logo Player_ID será removido
# Video_Avaiable desnecessário, podemos dropar

spida_all.drop(['Player_ID', 'VIDEO_AVAILABLE'], axis=1, inplace=True)

### Transformação de valores em colunas

In [19]:
spida_all['SEASON_ID'].value_counts()

SEASON_ID
22017    79
22018    77
22019    69
22022    68
22021    67
22023    55
22020    53
Name: count, dtype: int64

In [20]:
#Em todos os anos tem um 2 na frente dos anos, removendo...
spida_all['SEASON_ID'] = spida_all['SEASON_ID'].str.replace('^2', '', regex=True)
# ^2 É um método regex onde o acento circunflexo indica que é no inicio da string.

In [21]:
spida_all['SEASON_ID'].value_counts()

SEASON_ID
2017    79
2018    77
2019    69
2022    68
2021    67
2023    55
2020    53
Name: count, dtype: int64

In [22]:
spida_all['GAME_DATE'] = pd.to_datetime(spida_all['GAME_DATE'], format='%b %d, %Y')

In [23]:
spida_all.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 468 entries, 0 to 467
Data columns (total 25 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   SEASON_ID   468 non-null    object        
 1   Game_ID     468 non-null    object        
 2   GAME_DATE   468 non-null    datetime64[ns]
 3   MATCHUP     468 non-null    object        
 4   WL          468 non-null    object        
 5   MIN         468 non-null    int64         
 6   FGM         468 non-null    int64         
 7   FGA         468 non-null    int64         
 8   FG_PCT      468 non-null    float64       
 9   FG3M        468 non-null    int64         
 10  FG3A        468 non-null    int64         
 11  FG3_PCT     468 non-null    float64       
 12  FTM         468 non-null    int64         
 13  FTA         468 non-null    int64         
 14  FT_PCT      468 non-null    float64       
 15  OREB        468 non-null    int64         
 16  DREB        468 non-null  

In [24]:
#Criando uma coluna para o mês e o ano com base na coluna de data

spida_all['month'] = spida_all['GAME_DATE'].dt.month
spida_all['year'] = spida_all['GAME_DATE'].dt.year

In [27]:
# Agora temos duas medições de ano, uma pelo SEASON_ID e outra pelo ano (year). Porém a SEASON_ID muitas vezes traz jogos do ano posterior, pois este codigo se trata a temporada por inteiro (Ex: 2018-2019).
# Renomeando os valores de SEASON_ID
spida_all['SEASON_ID'] = spida_all['SEASON_ID'].str.replace('2023', '23-24')
spida_all['SEASON_ID'] = spida_all['SEASON_ID'].str.replace('2022', '22-23')
spida_all['SEASON_ID'] = spida_all['SEASON_ID'].str.replace('2021', '21-22')
spida_all['SEASON_ID'] = spida_all['SEASON_ID'].str.replace('2020', '20-21')
spida_all['SEASON_ID'] = spida_all['SEASON_ID'].str.replace('2019', '19-20')
spida_all['SEASON_ID'] = spida_all['SEASON_ID'].str.replace('2018', '18-19')
spida_all['SEASON_ID'] = spida_all['SEASON_ID'].str.replace('2017', '17-18')

In [51]:
#Faremos agora 2 novas colunas a partir de MATCHUP. Uma será o time em que o jogador participava naquele jogo e a outra seu time oponente.
'''
spida_all[['Team', 'Opponent']] = spida_all['MATCHUP'].str.split('vs.', expand=True)
spida_all[['Team2', 'Opponent']] = spida_all['MATCHUP'].str.split('@', expand=True)
'''
#Não funcionou. A tabela Matchup tem dois separadores diferentes, vs quando joga em casa e @ quando joga fora de casa.

In [59]:
#Criando uma função para os dois casos
def split_matchup(row):
    if ' vs. ' in row:
        return row.split(' vs. ')
    elif ' @ ' in row:
        return row.split(' @ ')
    return [row, '']

In [60]:
#Agora criando duas colunas e separando com base na função anterior
spida_all[['Team', 'Opponent']] = spida_all['MATCHUP'].apply(lambda x: pd.Series(split_matchup(x)))

# O método .apply() permite aplicar uma função a cada elemento de uma Series (ou DataFrame).
# lambda x: ... define uma função anônima (lambda) que toma um argumento x (que representa cada valor na coluna "MATCHUP").
# split_matchup(x) aplica a função split_matchup ao valor x.
# pd.Series(...) transforma o resultado de split_matchup(x) em uma Series do pandas. Isso é necessário porque a função -
# - split_matchup retorna uma lista, e a conversão para Series permite expandir os elementos da lista em colunas separadas.

In [ ]:
#Conferindo que a conversão se sucedeu corretamente
spida_all['Team'].value_counts()

In [65]:
#Criando uma Nova coluna para informar se os jogos foram em casa ou fora:
def determinar_tipo_jogo(matchup):
    if ' vs. ' in matchup:
        return 'home'
    elif ' @ ' in matchup:
        return 'away'
    else:
     return None

In [67]:
spida_all['type_of_game'] = spida_all['MATCHUP'].apply(lambda x: determinar_tipo_jogo(x))

In [ ]:
#Conferindo que a conversão se sucedeu corretamente
spida_all['type_of_game'].value_counts()

In [ ]:
spida_all

## Carregamento

In [71]:
# Comando de abertura para Google Drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [72]:
#Salvar Dataframe no Drive
spida_all.to_csv('/content/drive/MyDrive/Mestrado/dmitchell.csv')

## Análises


In [75]:
#Na sua carreira, quantas vitórias Donovan teve, quantas derrotas e o índice
spida_all['WL'].value_counts()

donovan_mitchell = {
    'Vitórias': 298,
    'Derrotas': 170
}

In [76]:
#Quantas vitórias no Cavs e quantas vitórias no Utah?
count_cle = spida_all.loc[(spida_all['Team'] == 'CLE') & (spida_all['WL'] == 'W'), 'WL'].count()
count_uta = spida_all.loc[(spida_all['Team'] == 'UTA') & (spida_all['WL'] == 'W'), 'WL'].count()

# .loc é usado principalmente para acessar um grupo de linhas e colunas por rótulos ou uma condição booleana.

In [77]:
#Adicionando no dicionário
donovan_mitchell['vitórias no Cavs'] = count_cle
donovan_mitchell['vitórias no Jazz'] = count_uta

In [78]:
donovan_mitchell

{'Vitórias': 298,
 'Derrotas': 170,
 'vitórias no Cavs': 80,
 'vitórias no Jazz': 218}

In [79]:
#Total de jogos no cavs, total de jogos no Utah e o índice de vitórias
total_cle_games = len(spida_all[spida_all['Team'] == 'CLE'])
total_uta_games = len(spida_all[spida_all['Team'] == 'UTA'])

In [80]:
#Adicionando no dicionário
donovan_mitchell['total de jogos no Cavs'] = total_cle_games
donovan_mitchell['total de jogos no Jazz'] = total_uta_games

In [82]:
win_pct_cle = (count_cle / total_cle_games) * 100
win_pct_uta = (count_uta / total_uta_games) * 100

In [83]:
#Adicionando no dicionário
donovan_mitchell['Porcentagem de vitória: Cavs'] = win_pct_cle
donovan_mitchell['Porcentagem de vitória: Jazz'] = win_pct_uta

In [84]:
donovan_mitchell

{'Vitórias': 298,
 'Derrotas': 170,
 'vitórias no Cavs': 80,
 'vitórias no Jazz': 218,
 'total de jogos no Cavs': 123,
 'total de jogos no Jazz': 345,
 'Porcentagem de vitória: Cavs': 65.04065040650406,
 'Porcentagem de vitória: Jazz': 63.18840579710145}

In [86]:
#Diminuindo casas decimais padrão no dicionário
for key in donovan_mitchell.keys():
    if key.startswith('Porcentagem'):
        donovan_mitchell[key] = f"{donovan_mitchell[key]:.2f}"

In [87]:
donovan_mitchell

{'Vitórias': 298,
 'Derrotas': 170,
 'vitórias no Cavs': 80,
 'vitórias no Jazz': 218,
 'total de jogos no Cavs': 123,
 'total de jogos no Jazz': 345,
 'Porcentagem de vitória: Cavs': '65.04',
 'Porcentagem de vitória: Jazz': '63.19'}

In [90]:
#Avaliando tabela mais a fundo (com apenas duas casas decimais)
spida_all.describe().round(2)

GAME_DATE     MIN     FGM     FGA  FG_PCT    FG3M  \
count                            468  468.00  468.00  468.00  468.00  468.00   
mean   2020-11-05 00:12:18.461538560   34.22    8.84   19.64    0.45    2.96   
min              2017-10-18 00:00:00   11.00    1.00    5.00    0.08    0.00   
25%              2019-01-10 12:00:00   32.00    6.75   16.00    0.36    2.00   
50%              2021-01-09 00:00:00   35.00    9.00   20.00    0.46    3.00   
75%              2022-10-30 18:00:00   37.00   11.00   23.00    0.53    4.00   
max              2024-04-12 00:00:00   50.00   22.00   36.00    0.80    9.00   
std                              NaN    5.07    3.30    5.36    0.12    1.76   

         FG3A  FG3_PCT     FTM     FTA  FT_PCT    OREB    DREB     REB  \
count  468.00   468.00  468.00  468.00  468.00  468.00  468.00  468.00   
mean     8.09     0.36    4.20    4.98    0.76    0.81    3.47    4.28   
min      1.00     0.00    0.00    0.00    0.00    0.00    0.00    0.00   
25%      6.00     0.25    2.00    2.00    0.67    0.00    2.00    3.00   
50%      8.00     0.36    4.00    4.00    0.86    1.00    3.00    4.00   
75%     10.00     0.46    6.00    7.00    1.00    1.00    5.00    6.00   
max     18.00     1.00   20.00   25.00    1.00    5.00   12.00   13.00   
std      2.89     0.18    3.11    3.56    0.31    0.98    2.02    2.22   

          AST     STL     BLK     TOV      PF     PTS  PLUS_MINUS   month  \
count  468.00  468.00  468.00  468.00  468.00  468.00      468.00  468.00   
mean     4.65    1.38    0.33    2.77    2.48   24.83        4.42    5.99   
min      0.00    0.00    0.00    0.00    0.00    2.00      -30.00    1.00   
25%      3.00    1.00    0.00    2.00    2.00   19.00       -5.00    2.00   
50%      5.00    1.00    0.00    3.00    2.00   25.00        5.00    3.00   
75%      6.00    2.00    1.00    4.00    3.00   31.00       15.00   11.00   
max     13.00    5.00    3.00    9.00    5.00   71.00       43.00   12.00   
std      2.31    1.14    0.56    1.66    1.32    9.06       13.14    4.53   

          year  
count   468.00  
mean   2020.39  
min    2017.00  
25%    2019.00  
50%    2021.00  
75%    2022.00  
max    2024.00  
std       2.08

In [92]:
# Dias que pontuou o mínimo da carreira
spida_all[spida_all['PTS'] == 2]

SEASON_ID     Game_ID  GAME_DATE      MATCHUP WL  MIN  FGM  FGA  FG_PCT  \
418     17-18  0021700793 2018-02-05    UTA @ NOP  W   19    1    6   0.167   
463     17-18  0021700062 2017-10-25    UTA @ PHX  L   13    1    7   0.143   
465     17-18  0021700036 2017-10-21  UTA vs. OKC  W   22    1    7   0.143   
466     17-18  0021700023 2017-10-20    UTA @ MIN  L   20    1    7   0.143   

     FG3M  FG3A  FG3_PCT  FTM  FTA  FT_PCT  OREB  DREB  REB  AST  STL  BLK  \
418     0     2      0.0    0    0     0.0     0     0    0    3    1    0   
463     0     2      0.0    0    0     0.0     0     1    1    1    1    1   
465     0     3      0.0    0    0     0.0     0     2    2    6    1    0   
466     0     2      0.0    0    1     0.0     0     0    0    2    0    2   

     TOV  PF  PTS  PLUS_MINUS  month  year Team Opponent Type of game  \
418    3   0    2          10      2  2018  UTA      NOP         away   
463    3   3    2          -2     10  2017  UTA      PHX         away   
465    0   4    2           3     10  2017  UTA      OKC         home   
466    1   2    2           3     10  2017  UTA      MIN         away   

    type_of_game  
418         away  
463         away  
465         home  
466         away